examples/workflow-standalone/src/main.ts

```diff
@@ -16,45 +16,67 @@
+    ActionHandlerRegistry,
     EnableToolPaletteAction,
     GLSPDiagramServer,
-    InitializeClientSessionAction,
     RequestTypeHintsAction
 } from '@eclipse-glsp/client';
-import { ApplicationIdProvider, BaseJsonrpcGLSPClient, JsonrpcGLSPClient } from '@eclipse-glsp/protocol';
+import {
+    ApplicationIdProvider,
+    BaseJsonrpcGLSPClient,
+    GLSPClient,
+    InitializeResult,
+    JsonrpcGLSPClient
+} from '@eclipse-glsp/protocol';
 import { join, resolve } from 'path';
 import { IActionDispatcher, RequestModelAction, TYPES } from 'sprotty';
 
 import createContainer from './di.config';
 
 const port = 8081;
 const id = 'workflow';
+const diagramType = 'workflow-diagram';
 const websocket = new WebSocket(`ws://localhost:${port}/${id}`);
-const container = createContainer();
 
 const loc = window.location.pathname;
 const currentDir = loc.substring(0, loc.lastIndexOf('/'));
 const examplePath = resolve(join(currentDir, '..', 'app', 'example1.wf'));
+const clientId = ApplicationIdProvider.get() + '_' + examplePath;
 
+const container = createContainer();
 const diagramServer = container.get<GLSPDiagramServer>(TYPES.ModelSource);
-diagramServer.clientId = ApplicationIdProvider.get() + '_' + examplePath;
-
-const actionDispatcher = container.get<IActionDispatcher>(TYPES.IActionDispatcher);
+diagramServer.clientId = clientId;
 
 websocket.onopen = () => {
     const connectionProvider = JsonrpcGLSPClient.createWebsocketConnectionProvider(websocket);
     const glspClient = new BaseJsonrpcGLSPClient({ id, connectionProvider });
-    diagramServer.connect(glspClient).then(client => {
-        client.initializeServer({ applicationId: ApplicationIdProvider.get() });
-        actionDispatcher.dispatch(new InitializeClientSessionAction(diagramServer.clientId));
-        actionDispatcher.dispatch(new RequestModelAction({
-            sourceUri: `file://${examplePath}`,
-            diagramType: 'workflow-diagram'
-        }));
-        actionDispatcher.dispatch(new RequestTypeHintsAction('workflow-diagram'));
-        actionDispatcher.dispatch(new EnableToolPaletteAction());
-    });
+    initialize(glspClient);
 };
 
+async function initialize(client: GLSPClient): Promise<void> {
+    await diagramServer.connect(client);
+    const result = await client.initializeServer({ applicationId: ApplicationIdProvider.get(), protocolVersion: GLSPClient.protocolVersion });
+    await configureServerActions(result);
+
+    const actionDispatcher = container.get<IActionDispatcher>(TYPES.IActionDispatcher);
+
+    await client.initializeClientSession({ clientSessionId: diagramServer.clientId, diagramType });
+    actionDispatcher.dispatch(new RequestModelAction({
+        sourceUri: `file://${examplePath}`,
+        diagramType
+    }));
+    actionDispatcher.dispatch(new RequestTypeHintsAction(diagramType));
+    actionDispatcher.dispatch(new EnableToolPaletteAction());
+}
+
+async function configureServerActions(result: InitializeResult): Promise<void> {
```

Comment:
Shouldn't this function and maybe even the entire `initialize` function be generic enough to push it into the generic base framework?

Reply:
Good catch. I don't think the `initialize` function itself is generic enough but the `configureServerActions` definitly is. I moved it into the base framework and reused it both here and in the glsp-theia-integration